build-system/compile/json-schema/index.js

```diff
@@ -182,42 +182,209 @@ function transformAjvCode(code, scope, config) {
   const findProperty = (properties, name) =>
     properties.find((node) => isObjectProperty(node, name));
 
+  /**
+   * @param {babel.NodePath<babel.types.MemberExpression>} memberExpression
+   * @return {null | undefined | number | string}
+   */
+  function evaluatePropertyKey(memberExpression) {
+    const property = memberExpression.get('property');
+    return property.isIdentifier()
+      ? property.node.name
+      : property.evaluate().value;
+  }
+
+  const schemaIdRegexp = /^schema[0-9]*$/;
+
+  /**
+   * @param {babel.NodePath<any>} path
+   * @param {string[]} keys
+   * @return {boolean}
+   */
+  function isMemberExpressionLeftwards(path, keys) {
+    let i = keys.length - 1;
+    while (true) {
+      if (path.isMemberExpression()) {
+        const key = evaluatePropertyKey(path);
+        if (key !== keys[i--]) {
+          return false;
+        }
+        path = path.get('object');
+      } else if (path.isVariableDeclarator()) {
+        const init = path.get('init');
```

Comment:
Is this branch actually needed? Otherwise this looks just like Babel's [`.matchesPattern()`](https://github.com/babel/babel/blob/df5c1ebb45040eff6c7146b7bee3a5142edc6223/packages/babel-traverse/src/path/introspection.ts#L17-L30)

Reply:
Unfortunately `matchesPattern` won't follow the binding. We're matching this case:

```js
const func0 = Object.prorotype.hasOwnProperty;
func0.call();
```